### ISSUE TITLE
Provide relative path when using File Dialog box

### ISSUE THREAD

**Issue by lie112** [APSIMInitiative/ApsimX#5781]
Is it possible to return the selected file path relative to the apsimx file when using the File Dialog box in Property Presenter? 
Saving full paths of files makes sharing project folders and apsimx files between users difficult.
If the file is outside the project folder then it would revert to full path.
Happy to wait for .Net Core Path.GetRelativePath
Checking if this will have carryover implications for people opening files, but I know the manual entry of just the filename and extension works

**Comment by hol430** [comment]
Yeah using relative paths is preferable for the reasons you listed. Some components (weather, excelinput, ...) already handle this internally but it might be useful to have a more consistent approach. Worth noting that not all paths can be expressed as relative paths (e.g. in windows a file on a different drive). So there would still need to be some capability for using absolute paths as a fallback. Shouldn't need to wait for .net core - we already have a `PathUtilities.GetRelativePath()`. The only problem I see is that we'd need to check every model which uses DisplayType.FileName(s), to ensure that it will work with relative paths as well as absolute.

**Comment by hol430** [comment]
The problem with doing this at the UI level is that the model will still need to handle relative/absolute path resolution when the simulation is run. If you look at the implementation in the weather component (FullFileName property), it will automatically handle this conversion and will attempt to store the file name as a relative path. That being said, I don't think we don't really want to duplicate this in every component which references an external file. So I'm not sure what the best solution is. I agree the current state of things is less than ideal.

Another (smaller) problem is that the file dialog doesn't have the context to determine a relative path, as it doesn't know about the .apsimx file or its file path.

**Comment by lie112** [comment]
Is this the task of the property presenter when the value is returned from the file dialog before deciding what to save? The property presenter should know paths. Given you can manually provide either the full path or the relative path in the property presenter and these both work in the simulation, I figured that no model would have any problem, but I may be missing some functionality implemented in other models.

**Comment by hol430** [comment]
Having this occur in the property presenter sounds like a good idea. I just checked and the only models using `DisplayType.FileName` are CLEM models, so you actually wouldn't need to check for compatibility in other existing models like weather, excel input, etc.
